### PR TITLE
[Customer Center] Improve subscription details view

### PR DESCRIPTION
**Pull request by codykerns**
The subscription details view could take advantage of more available space

- expands details view
- increases font sizes
- wraps in a scrollview for smaller screen sizes
- softens shadow
- have buttons fill available space, limit width via container views

| New           | Old           |
|:-------------:|:-------------:|
| New Version | Old Version |

### CODE REVIEW

File: RevenueCatUI/CustomerCenter/Views/ManageSubscriptionsView.swift [hunk 1]
```diff
@@ -172,78 +175,89 @@ struct SubscriptionDetailsView: View {
...
                 }
             }
-        }.padding()
-            .padding(.horizontal)
-            .background(Color(UIColor.tertiarySystemBackground))
-            .cornerRadius(20)
-            .shadow(color: Color.black.opacity(0.2), radius: 4)
+
+        }
+        .padding(24.0)
```
Comment (codykerns) [inline RevenueCatUI/CustomerCenter/Views/ManageSubscriptionsView.swift:256]: this is entirely subjective, although the corner radius and shadow details below were as well - not sure if we need to pull this out of here

File: RevenueCatUI/CustomerCenter/ManageSubscriptionsButtonStyle.swift [hunk 2]
```diff
@@ -33,7 +33,7 @@ struct ManageSubscriptionsButtonStyle: PrimitiveButtonStyle {
         let textColor = color(from: appearance.buttonTextColor, for: colorScheme)
 
         Button(action: { configuration.trigger() }, label: {
-            configuration.label.frame(width: 300)
+            configuration.label.frame(maxWidth: .infinity)
```
Comment (aboedo) [inline RevenueCatUI/CustomerCenter/ManageSubscriptionsButtonStyle.swift:36]: 👍

**Review by aboedo — APPROVED** [review]
😻

**Review by tonidero — APPROVED** [review]
Looks great!

File: RevenueCatUI/CustomerCenter/ManageSubscriptionsButtonStyle.swift (hunk 2 above)
Comment (JayShortway) [reply RevenueCatUI/CustomerCenter/ManageSubscriptionsButtonStyle.swift:36]: I was struggling with this, as a full width card didn't look good with the text size I had. Looks much better with the larger text! 🙌

File: RevenueCatUI/CustomerCenter/Views/ManageSubscriptionsView.swift [hunk 3]
```diff
@@ -354,6 +369,24 @@ struct ManageSubscriptionsView_Previews: PreviewProvider {
...
+struct SubscriptionDetailsView_Previews: PreviewProvider {
+
+    static var previews: some View {
+        SubscriptionDetailsView(
+            subscriptionInformation: CustomerCenterConfigTestData.subscriptionInformationMonthlyRenewing,
+            localization: CustomerCenterConfigTestData.customerCenterData.localization,
+            refundRequestStatusMessage: "Success"
+        )
+        .previewDisplayName("Subscription Details - Monthly")
+        .padding()
```
Comment (JayShortway) [inline RevenueCatUI/CustomerCenter/Views/ManageSubscriptionsView.swift:386]: Nice to have a separate preview!

**Review by JayShortway — APPROVED** [review]
Nice upgrade!

**Pull request merged**